File: backend/app/models/screening.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional
from datetime import datetime

from .document import OCRResult, MRZResult
from .validation import DocumentValidationResult
from .forensic import ForensicAnalysisResult
from .risk import RiskAssessmentResult

class FaceVerificationResult(BaseModel):
    similarity: float = 0.0 # Cosine similarity (0-1)
    status: str = "UNABLE_TO_VERIFY" # MATCH, REVIEW_REQUIRED, NO_FACE_DETECTED, UNABLE_TO_VERIFY
    confidence: float = 0.0
    detected_in_document: bool = False
    detected_in_reference: bool = False
    details: str = "No reference photo supplied."

class WatchlistMatchResult(BaseModel):
    matched: bool = False
    document_number: Optional[str] = None
    status: str = "CLEAN"
    reason: Optional[str] = None
    disclaimer: str = "Demo database — not connected to government systems."

class ScreeningRecord(BaseModel):
    screening_id: str
    filename: str
    file_path: str
    document_type: str
    created_at: str
    status: str = "COMPLETED" # PROCESSING, COMPLETED, FAILED
    
    # Modules Data
    ocr_result: Optional[OCRResult] = None
    mrz_result: Optional[MRZResult] = None
    validation_result: Optional[DocumentValidationResult] = None
    forensic_result: Optional[ForensicAnalysisResult] = None
    face_result: Optional[FaceVerificationResult] = None
    watchlist_result: Optional[WatchlistMatchResult] = None
    
    # Aggregated Engine
    risk_assessment: Optional[RiskAssessmentResult] = None
    ai_summary: Optional[str] = None
    
    # Metadata & URLs
    original_image_url: str
    annotated_image_url: Optional[str] = None
    report_pdf_url: Optional[str] = None
    processing_time_ms: int = 0

File: frontend/src/types/screening.ts
import { OCRResult, MRZResult } from './document';
import { ForensicAnalysisResult } from './forensic';
import { RiskAssessmentResult } from './risk';

export interface RuleCheckResult {
  rule_name: string;
  passed: boolean;
  severity: 'LOW' | 'MEDIUM' | 'HIGH';
  message: string;
  field?: string;
}

export interface DocumentValidationResult {
  document_type: string;
  valid: boolean;
  checks: RuleCheckResult[];
  total_checks: number;
  passed_checks: number;
  failed_checks: number;
}

export interface FaceVerificationResult {
  similarity: number;
  status: 'MATCH' | 'REVIEW_REQUIRED' | 'NO_FACE_DETECTED' | 'UNABLE_TO_VERIFY';
  confidence: number;
  detected_in_document: boolean;
  detected_in_reference: boolean;
  details: string;
}

export interface WatchlistMatchResult {
  matched: boolean;
  document_number?: string;
  status: string;
  reason?: string;
  disclaimer: string;
}

export interface ScreeningRecord {
  screening_id: string;
  filename: string;
  file_path: string;
  document_type: string;
  created_at: string;
  status: 'PROCESSING' | 'COMPLETED' | 'FAILED';
  
  ocr_result?: OCRResult;
  mrz_result?: MRZResult;
  validation_result?: DocumentValidationResult;
  forensic_result?: ForensicAnalysisResult;
  face_result?: FaceVerificationResult;
  watchlist_result?: WatchlistMatchResult;
  
  risk_assessment?: RiskAssessmentResult;
  ai_summary?: string;
  
  original_image_url: string;
  annotated_image_url?: string;
  report_pdf_url?: string;
  processing_time_ms: number;
}

export interface DemoScenario {
  id: string;
  title: string;
  description: string;
  expected_risk: string;
  badge: string;
}

export interface DocumentComparisonResult {
  comparison_id: string;
  difference_score: number;
  changed_regions_count: number;
  changed_regions: Array<{
    id: string;
    region: { x: number; y: number; width: number; height: number };
    severity: 'LOW' | 'MEDIUM' | 'HIGH';
    label: string;
    area_pixels: number;
  }>;
  annotated_diff_image_url: string;
  original_image_url: string;
  presented_image_url: string;
}
